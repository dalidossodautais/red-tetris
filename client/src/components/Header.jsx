import React from "react";
import AppBar from "@material-ui/core/AppBar";
import { makeStyles } from "@material-ui/core";
import Spacing from "./Spacing";
import MainTitle from "./texts/MainTitle";

function useClasses() {
  return makeStyles((theme) => ({
    appBar: {
      alignItems: "center",
      backgroundColor: theme.primaryBackgroundColor,
      color: theme.primaryTextColor,
      display: "flex",
      height: "8vh",
      minHeight: "48px",
      justifyContent: "center",
      userSelect: "none",
      width: "100%",
    },
  }))();
}

export default function Header({ children, onClick }) {
  const classes = useClasses();
  return (
    <>
      <AppBar className={classes.appBar} onClick={onClick}>
        <MainTitle>{children}</MainTitle>
      </AppBar>
      <Spacing height="10vh" minHeight="60px" width="100%" />
    </>
  );
}
