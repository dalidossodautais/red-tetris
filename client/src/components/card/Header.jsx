import React from "react";
import styled from "styled-components";
import AppBar from "@material-ui/core/AppBar";
import { makeStyles } from "@material-ui/core/styles";
import Toolbar from "@material-ui/core/Toolbar";
import Icon from "../Icon";
import TextField from "../inputs/TextField";

export const CategoryContainer = styled.span.attrs((props) => ({
  color: props.theme.primaryTextColor,
}))`
  align-items: center;
  color: ${(props) => props.color};
  display: flex;
  justify-content: space-between;
  user-select: none;
`;

const Title = styled.h2``;

function useClasses() {
  return makeStyles((theme) => ({
    appBar: {
      backgroundColor: theme.primaryBackgroundColor,
    },
    toolbar: {
      alignItems: "center",
      display: "flex",
      height: "80px",
      justifyContent: "space-between",
    },
  }))();
}

export default function CardHeader({
  iconName,
  label,
  onChange,
  title,
  value,
  width = "180px",
}) {
  const classes = useClasses();
  return (
    <AppBar className={classes.appBar} position="static">
      <Toolbar className={classes.toolbar}>
        <CategoryContainer>
          <Icon className={classes.icon} marginRight="15px" type={iconName} />
          <Title>{title}</Title>
        </CategoryContainer>
        <TextField
          color="secondary"
          hide={!onChange}
          label={label}
          onChange={onChange}
          value={value}
          variant="outlined"
          width={width}
        />
      </Toolbar>
    </AppBar>
  );
}
