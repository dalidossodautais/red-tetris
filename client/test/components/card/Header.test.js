import React from "react";
import renderer from "react-test-renderer";
import Header from "../../../src/components/card/Header";

describe("Header", () => {
  test("Standard", () => {
    const component = renderer.create(<Header />);
    // rendering
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
