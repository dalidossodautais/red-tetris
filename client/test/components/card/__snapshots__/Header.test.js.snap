// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Standard 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-1 MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
  >
    <span
      className="sc-bdfBwQ dYepCs"
    >
      <svg
        aria-describedby={null}
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-icon-3"
        focusable="false"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title=""
        viewBox="0 0 24 24"
      >
        <path
          d="M22.7 19l-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
        />
      </svg>
      <h2
        className="sc-gsTCUz"
      />
    </span>
  </div>
</header>
`;

exports[`Header Standard 2`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-1 MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
  >
    <span
      className="sc-bdfBwQ dYepCs"
    >
      <svg
        aria-describedby={null}
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-icon-3"
        focusable="false"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title=""
        viewBox="0 0 24 24"
      >
        <path
          d="M22.7 19l-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
        />
      </svg>
      <h2
        className="sc-gsTCUz"
      />
    </span>
  </div>
</header>
`;
